drivers/include/sx1280.h

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sx1280 LoRa radio driver
+ * @ingroup     drivers_netdev
+ * @brief       Driver for the sx1280 LoRa radio device
+ *
+ * @{
+ *
+ * @file
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ */
+
+#ifndef sx1280_H
+#define sx1280_H
+
+#include <assert.h>
+
+#include "sx1280_driver/src/sx1280_hal.h"
+#include "sx1280_driver/src/sx1280.h"
+#include "smtc_ral/src/ral.h"
+#include "smtc_ral/src/ral_defs.h"
+
+#include "net/netdev.h"
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    SX1280 device default configuration
+ * @{
+ */
+#define SX1280_CHANNEL_DEFAULT           (2403000000UL)          /**< Default channel frequency, 2403MHz */
+#define SX1280_RADIO_TX_POWER            (SX1280_PWR_MAX)        /**< Radio power in dBm */
+/** @} */
+
+/**
+ * * @note Forward declaration of the sx1280 device descriptor
+ */
+typedef struct sx1280 sx1280_t;
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    spi_t spi;                          /**< SPI device */
+    spi_mode_t spi_mode;                /**< SPI mode */
+    spi_clk_t spi_clk;                  /**< SPI clk */
+    gpio_t nss_pin;                     /**< SPI NSS pin */
+    gpio_t reset_pin;                   /**< Reset pin */
+    gpio_t busy_pin;                    /**< Busy pin */
+    gpio_t dio0_pin;                    /**< Dio0 pin */
+    gpio_t dio1_pin;                    /**< Dio1 pin */
+    gpio_t dio2_pin;                    /**< Dio2 pin */
+    gpio_t dio3_pin;                    /**< Dio3 pin */
```

Comment:
Can you check which `diox` pins are actually needed/mapped? Maybe we can remove `dio2`/`dio3`?